Sources/joyfill-components-swift/Components/Table/ViewTable.swift

```diff
@@ -612,51 +699,82 @@ extension ViewTable: UICollectionViewDelegate, UICollectionViewDataSource, UICol
     
     // MARK: CollectionView delegate method for cell selection at indexPath
     public func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
-        if tableDisplayMode == "readonly" {
-            
-        } else {
+        if tableDisplayMode != "readonly" {
             if indexPath.item == 0 {
                 if indexPath.section == selectedIndexPath {
                     let cell = collectionView.cellForItem(at: indexPath) as? CollectionViewCell
-                    cell?.selectionButton.setImage(UIImage(named: "selectButton"), for: .normal)
+                    cell?.selectionButton.setImage(UIImage(named: "unSelectButton"), for: .normal)
                     selectedIndexPath = nil
+                    moreButton.isHidden = true
+                    collectionView.reloadData()
                 } else {
                     let cell = collectionView.cellForItem(at: indexPath) as? CollectionViewCell
                     cell?.selectionButton.setImage(UIImage(named: "selectButton"), for: .normal)
+                    selectedIndexPath = indexPath.section
+                    cellSelectedIndexPath = indexPath
+                    moreButton.isHidden = false
                     collectionView.reloadData()
                 }
-                selectedIndexPath = indexPath.section
-                cellSelectedIndexPath = indexPath
-                moreButton.isHidden = false
-            } else {
-
+            }
+            if tableColumnType[indexPath.row] == "dropdown" {
+                let dropdownOptionArray: NSArray = ["Yes","No","N/A"]
+                let vc = CustomModalViewController()
+                vc.modalPresentationStyle = .overCurrentContext
+                vc.doneHide = "singleSelect"
```

Comment:
What is doneHide? Can we name this better

Reply:
Done
I renamed it as hideDoneButtonOnSingleSelect